src/playdate.ts

```diff
@@ -0,0 +1,80 @@
+import { spawnSync } from "child_process";
+import * as vscode from "vscode";
+
+export function compile(): boolean {
+    const ws = workspaceRoot();
+    const sdk = sdkPath();
+    const pdc = `${sdk}/bin/pdc`;
+
+    if (!(ws && sdk)) {
+        return false;
+    }
+
+    switch (process.platform) {
+    case "win32":
+        spawnSync("cmd", ["/c", pdc, `${ws}/source`, `${ws}/output.pdx`]);
```

Comment:
Probably worth checking the exit code for all of these commands to show proper feedback if they fail.